File: sprint2/By languages/Java/lesson10code1.java
public class Main {
    public static void main(String[] args) {
        func(1);
        func(2);
        func(3);
        sayHello("Гоша");
    }
    
    public static void func(int step) {
        System.out.println("Call me. Step: " + step);
    }
    
    public static void sayHello(String name) {
        System.out.println("Привет, " + name);
        printHoroscope(name.toUpperCase());
        System.out.println("Пока, " + name + ", хорошего дня!");
    }
    
    public static void printHoroscope(String name) {
        System.out.println(name + "! Сегодня подходящий день для прогулок в парке и изучения рекурсии");
    }
}

File: sprint2/By languages/C++/lesson10code1.cpp
#include <iostream>
#include <string>

void func(int step) {
    std::cout << "Call me. Step: " << step << std::endl;
}

void print_horoscope(std::string name) {
    std::cout << name << "! Сегодня подходящий день для прогулок в парке и изучения рекурсии" << std::endl;
}

void say_hello(std::string name) {
    std::cout << "Привет, " << name << std::endl;
    print_horoscope(name);
    std::cout << "Пока, " << name << ", хорошего дня!" << std::endl;
}

int main() {
    func(1);
    func(2);
    func(3);

    say_hello("Гоша");

    return 0;
}
